### DISCUSSION TITLE
Is it possible to as Tauri to build only to a specific architecture for android?

### DISCUSSION THREAD

**Discussion by charlesschaefer** [tauri-apps/tauri#10511] [Q&A] [answered]
I want to make tests in the final build of the app sometimes, but have to wait Tauri to build all  the architectures it has listed (i.e: i686, x86_64, aarch64, armv7, etc).

But I want to test only on my phone, that is an armv7. Is there any CLI options to tell Tauri to build only armv7?

I tried `npm run tauri android build --target=armv7`` but this doesn't work.

**Comment by FabianLars** [comment]
> I tried `npm run tauri android build --target=armv7`` but this doesn't work.

npm needs another `--` to actually forward the args to tauri so it's `npm run tauri android build -- --target armv7`

Edit: Forget the extra -- lol

**Reply by charlesschaefer** [reply]
Hi @FabianLars, thank you for your answer. I've tried this way on Windows' Powershell, but no success. Do you know if we need to do it differently on windows?

**Reply by FabianLars** [reply]
I am also on Windows + PowerShell and that's the command i use.

**Reply by charlesschaefer** [reply]
When I try `npm run tauri android build -- --target armv7` it shows the error:

"error: unexpected argument 'aarch64' found"

When I try `npm run tauri android build -- --target=armv7` it builds all the other architectures as well.

**Reply by FabianLars** [reply]
Hmm, that's weird. For me both commands behave the same and correctly.

Which npm version are you using? I have 10.8.1

Also, are we talking about the PowerShell version that's shipped with the os or `pwsh` which needs to be installed manually?

**Reply by charlesschaefer** [reply]
I'm using npm 10.8.2 and using the PowerShell that is shipped with Windows.

**Reply by charlesschaefer** [reply, marked as answer]
Oh, after trying some dumb ways to pass the target option, I was able to compile only one platform by doing:
`npm run tauri android build -- -- --target armv7`

I had to duplicate the double dash. That's weird.

**Reply by thautwarm** [reply]
For pnpm: `pnpm run tauri android build --target armv7`
